lib/middleware/routes-output/routes-output.js

```diff
@@ -67,12 +67,12 @@ const generateOutput = async (req, res) => {
     }
 
     const componentInstances = jp.query(pageInstance, '$..[?(@._type)]')
-    componentInstances.forEach(componentInstance => {
+    for await (let componentInstance of componentInstances) {
       const componentInstanceController = getInstanceController(componentInstance)
       if (componentInstanceController.preUpdateContents) {
-        componentInstance = componentInstanceController.preUpdateContents(componentInstance, userData, pageInstance)
+        componentInstance = await componentInstanceController.preUpdateContents(componentInstance, userData, pageInstance)
```

Comment:
Can remove unused assignment of componentInstance

Reply:
Removed